alma/lib/Utils/CustomerHelper.php

```diff
@@ -0,0 +1,199 @@
+<?php
+/**
+ * 2018-2022 Alma SAS
+ *
+ * THE MIT LICENSE
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+ * Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+ * WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ *
+ * @author    Alma SAS <contact@getalma.eu>
+ * @copyright 2018-2022 Alma SAS
+ * @license   https://opensource.org/licenses/MIT The MIT License
+ */
+
+namespace Alma\PrestaShop\Utils;
+
+use Alma\PrestaShop\Model\AddressData;
+use Alma\PrestaShop\Model\CarrierData;
+use Alma\PrestaShop\Model\CartData;
+use Cart;
+use Context;
+use Country;
+use Customer;
+use Order;
+
+if (!defined('_PS_VERSION_')) {
+    exit;
+}
+
+/**
+ * Class CustomerHelper.
+ *
+ * Customer Helper
+ */
+class CustomerHelper
+{
+    public function __construct(
+        Customer $customer,
+        Context $context,
+        Cart $cart
+    ) {
+        $this->customer = $customer;
+        $this->context = $context;
+        $this->cart = $cart;
+    }
+
+    /**
+     * Get array data customer
+     *
+     * @return array getData
+     */
+    public function getData()
+    {
+        $addressData = new AddressData($this->cart);
+
+        return [
+            'first_name' => $this->customer->firstname,
+            'last_name' => $this->customer->lastname,
+            'email' => $this->customer->email,
+            'birth_date' => $this->getBirthday(),
+            'addresses' => $this->getAddressesData(),
+            'phone' => $this->getPhone(),
+            'country' => $addressData->getBillingCountry(),
+            'county_sublocality' => null,
+            'state_province' => $addressData->getBillingStateProvince(),
+        ];
+    }
+
+    public function getWebsiteDetails()
+    {
+        $carrierData = new CarrierData($this->context);
```

Comment:
try to make all your instanciations (`new`) int constructor or before constructor is you don't plan to use context directly into your helper